### DISCUSSION TITLE
Help in resolving `BadInitialCandidatesWarning`

### DISCUSSION THREAD

**Discussion by shikhar-saxena** [meta-pytorch/botorch#1934] [Q&A] [answered]
I am getting the warning while using `qExpectedImprovement` Acquisition Function (for `q=1`) 
```
BadInitialCandidatesWarning: Unable to find non-zero acquisition function values - initial conditions are being selected randomly
```

Can someone tell me how to resolve this? I increased `num_restarts` (from 5 to 10) and `raw_samples` (from 512 to 1024) but that didn't help.

**Comment by saitcakmak** [comment]
How does your `best_f` value compare with `train_Y`? This warning would happen if the acquisition function is unable to generate posterior samples that are better than `best_f`.

**Reply by shikhar-saxena** [reply]
I'm taking `best_f = train_Y.max().item()`. 

> This warning would happen if the acquisition function is unable to generate posterior samples that are better than best_f.

So basically, my `best_f` value is too high for the algorithm to find some improvement with respect to it. Not sure how I can fix this. What does botorch do in this case btw?

I'm also using PI acquisition function (for this same setting). And PI doesn't give me this warning.

**Reply by Balandat** [reply, marked as answer]
This seems like a posterchild use case for the new `qLogEi`, which addresses these issues and is about to be merged in: https://github.com/pytorch/botorch/pull/1936

**Reply by shikhar-saxena** [reply]
Thanks. I tried running my code with `LogExpectedImprovement`, and it no longer gives the warning.

Can you please let me know when `qLogExpectedImprovement` will be released?

**Reply by Balandat** [reply]
Great to hear. It'll be in the next release. We don't have a specific schedule for this one - @sdaulton were you planning to get other things into a release before next week?

**Comment by jakobzeitler** [comment]
Thanks so much, very helpful! Using this right now.
